### ISSUE TITLE
AnalyticalTable: announce row data on selection via screen reader

### ISSUE THREAD

**Issue by durgaprasad1995** [SAP/ui5-webcomponents-react#6273]
### Bug Description

Accessibility Issue.

When the user selects and deselects the checkbox it should read the row details. 

Currently, it's reading only checked and unchecked.

**Observed Behavior:**

There is no information conveyed to the end user upon selecting or deselecting the rows present within the 'Items(xxxx)' table which is present within the 'Supplier' dialog box.

 

**Expected Behavior:**

State of the should be read out to screen reader users.
Providing state of the UI on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.
Here information as 'Checked/selected' should be read out when sub menu items are accessed.

State (expand/collapse) of the UI should be read out to screen reader users.

### Expected Behaviour

State of the should be read out to screen reader users.
Providing state of the UI on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.
Here information as 'Checked/selected' should be read out when sub menu items are accessed.

State (expand/collapse) of the UI should be read out to screen reader users.

### Steps to Reproduce

1.Go to analytics table (https://sap.github.io/ui5-webcomponents-react/v2/?path=/story/data-display-analyticaltable--default&args=selectionMode:Multiple) check any row 
2.Open Jaws and see whether it is announcing the selected row data or not
3.
...

### Log Output, Stack Trace or Screenshots

![image](https://github.com/user-attachments/assets/5d23cf9a-f2a3-4ba0-a2c4-6968bb9ff9e1)

### Priority

Very High

### UI5 Web Components Version

1.17.0

### Browser

Chrome

### Operating System

Windows

### Organization

SAP

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by durgaprasad1995** [comment]
Hi @Lukas742  there is no update since 2weeks

**Comment by durgaprasad1995** [comment]
Hi @Lukas742 i hope you got all the required information kindly let me know if anything

**Comment by Lukas742** [comment]
Hi @durgaprasad1995 

we will investigate this issue in the upcoming week at the latest.

**Comment by elenastoyanovaa** [comment]
Hello @Lukas742 ,

Perhaps the reporter should answer this question, or a specification should be checked?

Kind Regards,
Elena

**Comment by Lukas742** [comment]
@elenastoyanovaa Oops, I didn't mean to tag you. Sorry about that! I've now edited my post.

**Comment by durgaprasad1995** [comment]
> Hi @durgaprasad1995
> 
> could you please elaborate on what the expected output should be? Each cell has an [aria-selected](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected#grid) prop which should announce the selected state of the cell/row. Additionally, we're setting an `aria-label` explaining how a row can be selected or unselected depending on the state.
> 
> As reference you can find an example using a standard HTML grid table, where a basic select functionality is implemented, here: https://stackblitz.com/edit/github-jsdyp7?file=src%2FHTMLGridTable.tsx,src%2FApp.tsx,src%2Findex.css

Hi @Lukas742  It should read out the state and row details. In the given example it's announcing the selected and deselect and  which row it is in but not announcing the details it should announce the whole row selected details Ex:- Row 3 Description 3 like this 

please let me know if anything needed

**Comment by durgaprasad1995** [comment]
Adding more information on Expected Behaviour.

Each & every UI element should be read out on accessing with screen reader.

Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

**Comment by Lukas742** [comment]
Hi @durgaprasad1995 

I wasn't able to find specifications for this, so for this to be implemented this request would need to go through the Inner Source process and be synchronised with the SAP Central Design team first. Please raise a BLI to the central design team and include the ticket number here for reference.

In the meantime you can also implement this behavior yourself. I've prepared a StackBlitz example showing how row and cell props can be extended and also how you can implement live announcements for screen readers.

https://stackblitz.com/edit/ui5wcr-at-selection-live-announcement?file=src%2FApp.tsx,node_modules%2F%40ui5%2Fwebcomponents-base%2Fdist%2Futil%2FInvisibleMessage.d.ts

**Comment by durgaprasad1995** [comment]
Hi @Lukas742 Thanks a lot I have tested with the example it's working as expected please find the recordings and screen shot.

please proceed next step. let me know the fixed version

![image](https://github.com/user-attachments/assets/4d3ed37b-051a-47ae-b659-13d6fdcf004c)


https://github.com/user-attachments/assets/32039f66-d59b-40e7-9d8f-ffd9015af0e9

**Comment by Lukas742** [comment]
Hi @durgaprasad1995 

If you want this behavior to be integrated in the `AnalyticalTable`, please file the BLI:

> I wasn't able to find specifications for this, so for this to be implemented this request would need to go through the Inner Source process and be synchronised with the SAP Central Design team first. Please raise a BLI to the central design team and include the ticket number here for reference.

Otherwise, you can use the provided code snippet to implement it in your application.

**Comment by durgaprasad1995** [comment]
Okay can i know the process how to create the BLI

**Comment by durgaprasad1995** [comment]
> @durgaprasad1995 since this is a public repo, please contact me internally so I can forward you the links.

Okay @Lukas742 Thanks will contact
